<!DOCTYPE doctype html>

<html lang="en">

<meta content="" name="description"/>
<link href="../CLstyle.css" rel="stylesheet" type="text/css">
<link href="http://fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,600italic,700italic,800italic,400,300,600,700,800" rel="stylesheet" type="text/css">
</link></link></meta></head>

<div id = "banner">
<a href="http://cnr.lwlss.net">
<div style="display:inline-block; align: center;vertical-align: middle;" >
  <img src="../logo.png" style="height:150px; padding-right: 20px;">
</div>
<div style="display:inline-block;vertical-align: middle;">
  <h1 style="margin-top:-10; margin-bottom:-10;">Conor Lawless</h1>
  <h2 style="margin-top:-10; margin-bottom:-10;">Scientific Computing</h2>
  <h3 style="margin-top:-10; margin-bottom:-10;">http://cnr.lwlss.net</h3>
</div>
</a>
</div>


	<head>
    <title>Building consensus sequences</title>
    <meta charset='utf-8'>
	<meta content="" name="description"/>
	<link href="../CLstyle.css" rel="stylesheet" type="text/css">
	<link href="http://fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,600italic,700italic,800italic,400,300,600,700,800" rel="stylesheet" type="text/css">
	</link></link></meta>
	<script type="text/x-mathjax-config">MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}});</script>
	<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
	</head>

<div id = "banner">
<a href="http://cnr.lwlss.net">
<div style="display:inline-block; align: center;vertical-align: middle;" >
  <img src="../logo.png" style="height:150px; padding-right: 20px;">
</div>
<div style="display:inline-block;vertical-align: middle;">
  <h1 style="margin-top:-10; margin-bottom:-10;">Conor Lawless</h1>
  <h2 style="margin-top:-10; margin-bottom:-10;">Scientific Computing</h2>
  <h3 style="margin-top:-10; margin-bottom:-10;">http://cnr.lwlss.net</h3>
</div>
</a>
</div>



	<body>
	<header>
	<h1>Building consensus sequences</h1>
	<time datetime="2013-06-30T14:27:00Z">14:27 Sunday 30th June 2013</time>
	</header>
	<article>
	
	<h2>Introduction</h2>
	
	<p>In my <a href="../seqCoverage">last</a> <a href="../seqReads">two</a> posts I described aligning short reads (stored in <a href="http://en.wikipedia.org/wiki/FASTQ_format">.fastq</a> format) produced by a sequencing machine against a reference sequence and visualising read-depth coverage along the genome.  In the final post in this series of three, I will briefly describe the next step in the resequencing workflow: generating a <a href="http://en.wikipedia.org/wiki/Consensus_sequence">consensus genomic sequences</a> from read-alignment (<a href="http://www.broadinstitute.org/igv/BAM">.bam</a> files, such as those generated in my <a href="../seqCoverage">last post</a>).</p>
	
	<h2>Software & scripting to generate consensus</h2>
	
	<p>In the <a href="../seqReads">first</a> of these three posts, I described installing the <a href="http://samtools.sourceforge.net/">samtools</a> software, along with using scripting tools like Bash shell scripts and Windows batch files to automate analysis.  Here I will demonstrate the use <a href="http://samtools.sourceforge.net/">samtools</a> again, this time to call <a href="https://en.wikipedia.org/wiki/Single-nucleotide_polymorphism">SNP</a>s and (short) <a href="http://en.wikipedia.org/wiki/Indel">indel</a>s and to output <a href="http://en.wikipedia.org/wiki/Consensus_sequence">consensus sequence</a>s.  Again, I will use scripting to automate this process for each sample to be analyzed:</p>
	
	<script src="https://gist.github.com/CnrLwlss/5895260.js"></script>
	
	<p>The script above runs the <a href="http://samtools.sourceforge.net/">samtools</a> <a href="http://samtools.sourceforge.net/mpileup.shtml">mpileup</a> function, which generates output in the <a href="http://en.wikipedia.org/wiki/Variant_Call_Format">Variant Call Format (VCF)</a> by default.  In order to skip that stage and move directly to the consensus sequence in <a href="http://en.wikipedia.org/wiki/FASTQ_format">.fastq</a> format (sequence and quality measure for each of 17 chromosome "reads") the script above <a href="http://en.wikipedia.org/wiki/Pipeline_(Unix)">pipes</a> the output from samtools mpileup directly to two other tools which are part of the samtools software: bcftools and the perl script vcfutils.  In order to get the last step to work under windows you will need a working <a href="http://strawberryperl.com/">perl</a> environment and a copy of the <a href="https://github.com/samtools/samtools/blob/develop/bcftools/vcfutils.pl">vcfutils.pl</a> <a href="http://strawberryperl.com/">perl</a> script which you can get from the samtools <a href="https://github.com/samtools/samtools/blob/develop/bcftools/vcfutils.pl">github</a> page.  In any case there is an example snippet of the output for the first few bases from one chromosome from one yeast sample below:</p>

<pre>
@ref|NC_001133|
mcmcaccacacccACACACCCACACACCACACCACACACCACACCACACCCACACACACA
CATCCTAACACTACCCTAACACAGCCCTAATCTAACCCTGGCCAACCTGTCTCTCAACTT
ACCCTCCATTACCCTGCCTCCACTCGTTACCCTGTCCCATTCAACCATACCACTCCGAAC
CACCATCCATCCCTCTACTTACTACCACTCACCCACCGTTACCCTCCAATTACCCATATC
</pre>

<h2>Automatically comparing gene sequences</h2>
	<p>Accessing and comparing subsets of these consensus sequences (e.g. checking if the sequence at an open-reading frame associated with your favourite gene is the same in your sample as in some other reference genome) can be done programatically using the <a href="http://biopython.org/wiki/Main_Page">BioPython</a> library for <a href="http://www.python.org/">Python</a> (another excellent open-source programming language).  Below is an example <a href="http://www.python.org/">Python</a> script which demonstrates reading in a reference genome (e.g. the <a href="http://downloads.yeastgenome.org/sequence/S288C_reference/genome_releases/S288C_reference_genome_Current_Release.tgz">S288C reference genome</a> from <a href="http://yeastgenome.org/">SGD</a>) and comparing that with each of the sample sequences generated with samtools (see script above).  In particular, it only compares the sequences within the <a href="http://en.wikipedia.org/wiki/Open_reading_frame">Open Reading Frames</a> for specified genes.  I have assumed the same directory structure I presented in the <a href="../seqReads">first post</a>.  Note that lines preceded by '#' are comments, or notes to help humans understand the motivation for commands given to the computer.</p>
	
<script src="https://gist.github.com/CnrLwlss/5935118.js"></script>	

	<p>Finally, here is some example output from the script above.  I have specifically chosen the genes examined not to be particularly informative, but you can see that there is no difference between the <a href="http://www.yeastgenome.org/cgi-bin/locus.fpl?locus=brr1"><i>BRR1</i></a> <a href="http://www.yeastgenome.org/cgi-bin/getSeq?query=YPR057W&seqtype=ORF%20Genomic%20DNA&format=fasta">reference sequence</a> and sample sequence in this case, whereas a large section of the <a href="http://www.yeastgenome.org/cgi-bin/locus.fpl?locus=his3"><i>HIS3</i></a> gene is missing (functionally deleted).  Note that, depending on how your browser handles word-wrapping, you will have to scroll far to the right to see the full gene sequences on this page.</p>
	
<pre>
BRR1

Reference BRR1 sequence:
ATGAAAAGAGGAGAAAGCCAGGCGCCTGATGCCATCTTTGGTCAATCTCGTGCATTTGCACTATCTGATTCTTCAGTGAATCCTGATGTCATTGAATATCTTAAAAGTGTTAGGCAAGAAGCACTAAGAACCAACGCCATATCGATTAAGAATCATATGAATCTGCAAAAGAGGACGCGTCACAAATCAAGCATGTACGATGATGAGGATGAGGGGGCCCTTAAAAGGCACGCCATCTCGCCATCCTTGATTAGGCTTCAAAGGAATGTAGAAATATGGGTAAGGTGGTTTAACTCTGTGAAGGCGACGGTGTTGACTAATGCCTACGAGTTTACCGGTTATGAGGACGAAACACTGGATCTTTTATTGCTTTTCTTAAAGAATTATCTCGAAGATATGCCCAGCAAATGTACTACAGTTGAAAAAATTATAAGTGTCCTAAATCAGCATTCCTTTCCTGAGAAGGCAGAAGAGAAGGAAGAAAATCTTCAGATTGATGAGGAATGGGCGAAGAATATTTTGGTACGGCTGGAAAAAACCAAGATTGACAGTGTTGAGGATGTAAAAAAAGTAATCACTGAAGGAGATAAACATGAACTAGTTGGATACAACCAGTGGTTCCAATACCTTATAAACAATGAACCGCAGCATACTACTTTTCATGAAAAGATTACCTCTAAGCAACTTTGGGTTCTGATCAAGTATATGTCGAATACATGGATAAAAGAAATACACAAGAAAGGAAGGCATTATCGTCGGCTGCAAGATTGGCTATTCTACATACTGGTACATACACCCGAAAGAGTCACGGCAGAATATACAAGCATCTTGAGAGATCTTGGAAAGAAATGCCTTGAACTGATTCAAAAGAAGCCAGTTGAAGCACATGAGAATAAAATAACACTCCCGAAGGAGATGGCGGAATTGAATGTTGAGATACCAGCCGCGGTGGAGAATATGACGATAACTGAGCTGACAGTGTCTGTTATAGCAGTGAACTATGGTCAAAAAGACTTGATAGAATAA

SN7640178_10172_8183 Sample BRR1 sequence:
ATGAAAAGAGGAGAAAGCCAGGCGCCTGATGCCATCTTTGGTCAATCTCGTGCATTTGCACTATCTGATTCTTCAGTGAATCCTGATGTCATTGAATATCTTAAAAGTGTTAGGCAAGAAGCACTAAGAACCAACGCCATATCGATTAAGAATCATATGAATCTGCAAAAGAGGACGCGTCACAAATCAAGCATGTACGATGATGAGGATGAGGGGGCCCTTAAAAGGCACGCCATCTCGCCATCCTTGATTAGGCTTCAAAGGAATGTAGAAATATGGGTAAGGTGGTTTAACTCTGTGAAGGCGACGGTGTTGACTAATGCCTACGAGTTTACCGGTTATGAGGACGAAACACTGGATCTTTTATTGCTTTTCTTAAAGAATTATCTCGAAGATATGCCCAGCAAATGTACTACAGTTGAAAAAATTATAAGTGTCCTAAATCAGCATTCCTTTCCTGAGAAGGCAGAAGAGAAGGAAGAAAATCTTCAGATTGATGAGGAATGGGCGAAGAATATTTTGGTACGGCTGGAAAAAACCAAGATTGACAGTGTTGAGGATGTAAAAAAAGTAATCACTGAAGGAGATAAACATGAACTAGTTGGATACAACCAGTGGTTCCAATACCTTATAAACAATGAACCGCAGCATACTACTTTTCATGAAAAGATTACCTCTAAGCAACTTTGGGTTCTGATCAAGTATATGTCGAATACATGGATAAAAGAAATACACAAGAAAGGAAGGCATTATCGTCGGCTGCAAGATTGGCTATTCTACATACTGGTACATACACCCGAAAGAGTCACGGCAGAATATACAAGCATCTTGAGAGATCTTGGAAAGAAATGCCTTGAACTGATTCAAAAGAAGCCAGTTGAAGCACATGAGAATAAAATAACACTCCCGAAGGAGATGGCGGAATTGAATGTTGAGATACCAGCCGCGGTGGAGAATATGACGATAACTGAGCTGACAGTGTCTGTTATAGCAGTGAACTATGGTCAAAAAGACTTGATAGAATAA
Sample: SN7640178_10172_8183 Gene: BRR1
Number/percentage of differing nucleotides: 0 0.0
HIS3

Reference HIS3 sequence:
ATGACAGAGCAGAAAGCCCTAGTAAAGCGTATTACAAATGAAACCAAGATTCAGATTGCGATCTCTTTAAAGGGTGGTCCCCTAGCGATAGAGCACTCGATCTTCCCAGAAAAAGAGGCAGAAGCAGTAGCAGAACAGGCCACACAATCGCAAGTGATTAACGTCCACACAGGTATAGGGTTTCTGGACCATATGATACATGCTCTGGCCAAGCATTCCGGCTGGTCGCTAATCGTTGAGTGCATTGGTGACTTACACATAGACGACCATCACACCACTGAAGACTGCGGGATTGCTCTCGGTCAAGCTTTTAAAGAGGCCCTAGGGGCCGTGCGTGGAGTAAAAAGGTTTGGATCAGGATTTGCGCCTTTGGATGAGGCACTTTCCAGAGCGGTGGTAGATCTTTCGAACAGGCCGTACGCAGTTGTCGAACTTGGTTTGCAAAGGGAGAAAGTAGGAGATCTCTCTTGCGAGATGATCCCGCATTTTCTTGAAAGCTTTGCAGAGGCTAGCAGAATTACCCTCCACGTTGATTGTCTGCGAGGCAAGAATGATCATCACCGTAGTGAGAGTGCGTTCAAGGCTCTTGCGGTTGCCATAAGAGAAGCCACCTCGCCCAATGGTACCAACGATGTTCCCTCCACCAAAGGTGTTCTTATGTAG

SN7640178_10172_8183 Sample HIS3 sequence:
ATGACAGAGCAGAAAGCCCTAGTAAAGCGTATTACAAATGAAACCAAGATTCAGATTGCGATCTCTTTAAAGGGTGGTCCCCTAGCGATAGAGCACTCGATCTTCCCAGAAAAAGAGGCAGAAGCAGTAGCAGAACAGGCCACACAATCGCAAGTGATTAACGTCCACACAGGTATAGGGTTTCTGGACCATATGATACATGCTCTGGCCAAGCATTCCGGCTGGTCGCTAATCGTTGAGTGCATTGGTGACTTACACATAGACGACCATCACACCACTGAAGACTGCGGGATTGCTCTCGGTcaagcttttaaagagGcccTAGggnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnatcccGCATtttctcgaaagcttTGCAGAGGCTAGCAGAATTACCCTCCACGTTGATTGTCTGCGAGGCAAGAATGATCATCACCGTAGTGAGAGTGCGTTCAAGGCTCTTGCGGTTGCCATAAGAGAAGCCACCTCGCCCAATGGTACCAACGATGTTCCCTCCACCAAAGGTGTTCTTATGTAG
Sample: SN7640178_10172_8183 Gene: HIS3
Number/percentage of differing nucleotides: 189 28.5067873303
</pre>
	

	</article>
</body>	
</html>