<!DOCTYPE doctype html>

<html lang="en">

<meta content="" name="description"/>
<link href="../CLstyle.css" rel="stylesheet" type="text/css">
<link href="http://fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,600italic,700italic,800italic,400,300,600,700,800" rel="stylesheet" type="text/css">
</link></link></meta></head>
	<head>
    <title>Extracting read coverage per base</title>
    <meta charset='utf-8'>
	<meta content="" name="description"/>
	<link href="../CLstyle.css" rel="stylesheet" type="text/css">
	<link href="http://fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,600italic,700italic,800italic,400,300,600,700,800" rel="stylesheet" type="text/css">
	</link></link></meta>
	<script type="text/x-mathjax-config">MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}});</script>
	<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
	</head>

	<body>
	<header>
	<h1>Extracting read coverage per base</h1>
	<time datetime="2013-06-27T11:08:00Z">11:08 Thursday 27th June 2013</time>
	</header>
	<article>
	
	<h2>Introduction</h2>
	<p>In my <a href="../seqReads">last post</a> I described the start of a typical high-throughput sequencing data analysis workflow: how to align paired-end reads, recorded in <a href="http://en.wikipedia.org/wiki/FASTQ_format">.fastq</a> files, against a reference sequence.  The outputs from that workflow are <a href="http://www.broadinstitute.org/igv/BAM">.bam</a> files describing the alignment of reads against the reference.  Such files are useful in several different contexts; I have had a number of requests for help with extracting read coverage (number of reads per nucleotide for some or all of the genome) from .bam files and so this seems like a natural place to start a new post.</p>
	
	<a href="http://www.iontorrent.com/"><img src="http://singularityhub.com/wp-content/uploads/2010/12/IonTorrent_PGM.jpg" title="An IonTorrent machine, popular at Newcastle.  Can be used for doing ChIP-sequencing for example."></a>
	
	<p>Some reasons for being interested in coverage:</p>
	<ol>
		<li>Pilot sequencing runs: Have I gathered sufficient reads to observe <a href="https://en.wikipedia.org/wiki/Single-nucleotide_polymorphism">SNP</a>s, <a href="https://en.wikipedia.org/wiki/Deletion_(genetics)">deletions</a> or <a href="http://en.wikipedia.org/wiki/Insertion_(genetics)">insertions</a> along the sampled genome?  Sequencing experiments are often designed with a minimum required number of reads per base.  Trial sequencing runs can help determine the amount of sequencing time (and cost) required to achieve this minimum number of reads.</li>
		<li>Genome-wide quality control: Are there any specific areas of the genome where my sequencing is running into trouble?  Regions of the genome with highly repetitive DNA, aneuploidy or gross <a href="http://en.wikipedia.org/wiki/Chromosomal_rearrangement">chromosomal rearrangements</a> can be detected by scanning the coverage along chromosomes.</li>
		<li><a href="http://en.wikipedia.org/wiki/ChIP-sequencing">ChIP-seq</a> to identify protein-DNA interactions: the number of reads per base (coverage) is proportional to the affinity of the target protein for sequences along the genome.  In this case, estimating coverage is the primary purpose of the experiment.</li>
	</ol>
	
	</p>My <a href="../seqReads">previous post</a> includes a look at the text data contained in uncompressed alignment files (.sam files) and a <a href="http://samtools.sourceforge.net/SAM1.pdf">specification</a> document describing the meaning of the various sections</p>
	
	<h2>Software Installation</h2>
	
	<p>The <a href="http://www.bioconductor.org/packages/release/bioc/html/ShortRead.html">ShortRead</a> package for <a href="http://www.r-project.org/">R</a> is well suited to analysing coverage in .bam files (<a href="http://dx.doi.org/10.1093/bioinformatics/btp450">Morgan et al. (2009)</a>).  R is a powerful, open-source, statistical programming language with many interesting add-ons available.  R is extremely useful for a wide range of tasks, particularly reading in data from files and generating plots.  It is well worth spending time getting to grips with R and there are several excellent <a href="http://www.ncl.ac.uk/maths/rcourse/">courses</a> run here in Newcastle each year.  There are also online tutorials like <a href="http://www.cyclismo.org/tutorial/R/">this one</a> to get you started.</p>
	
	<h3>R</h3>
	<p>If you don't already have R installed on your machine, it's easy to just install it now.  R is open source and freely available online.  Note that you will need to have administrator rights on your machine.  You can download the relevant installation files for Windows or OSX <a href="http://cran.rstudio.com/">here</a>.  For Newcastle University users: R has recently been rolled out onto many of the university desktop machines.  If it's not installed on your machine, raising a request with your local IT support officer should get it sorted out.  For Ubuntu (Linux) users, you can install R on the command-line as follows:</p>
	
	<pre>
sudo apt-get update
sudo apt-get install r-base
   </pre>
   
   <p>Once R is installed, you can start it by typing "R" (followed by ENTER) at the command-line to get a text-only interface or navigate to an R shortcut for the <a href="http://en.wikipedia.org/wiki/Graphical_user_interface">GUI</a> which has a windowed interface (e.g. on Windows: Start->All Programs->R->Rx64 3.0.1).</p>
	
	<p>Regardless of which interface you prefer, commands are issued in the command window, predominantly by typing or pasting text.  Importantly, all commands are fully scriptable (can be written in a text file, saved, copied, edited, distributed etc.), which saves a huge amount of time, even in the short-term and makes analysis reproducible and allows it to be assessed by others.</p>
   
   <h3>ShortRead</h3>
   
   <p>Once R is installed, you need to install an add-on for R which can handle high-throughput sequencing data.  The <a href="http://www.bioconductor.org/packages/release/bioc/html/ShortRead.html">ShortRead</a> package on <a href="http://www.bioconductor.org/">bioconductor</a> is designed for this task.  To install it execute the following two lines (each followed by ENTER) in an R console:</p>
 
<pre>
source("http://bioconductor.org/biocLite.R")
biocLite("ShortRead") 
</pre>

<h2>Analysis</h2>

<h3>Human Chromosome 11</h3>

<p>Here is a simple example script which uses the R package ShortRead to visualise coverage.  It takes reads from a human genome (chromosome 11) which was sequenced as part of the <a href="http://www.1000genomes.org/">1,000 genomes</a> project.  You can download the .bam file <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data/HG00100/alignment/HG00100.chrom11.ILLUMINA.bwa.GBR.low_coverage.20130415.bam">here</a>.</p>

<p>To run the script yourself, place the .bam file into a directory on your computer and set the "working directory" in your R session to point to that directory.  For example, using the R GUI under Windows, click on the R Console window to bring it into focus and select File -> Change Dir to navigate to the directory containing the .bam file.  There are several ways to run the script, but probably the clearest is to save the script to a text file in the same directory.  Give it a name like "Coverage.R".  Then run it by typing the following command in the R console (followed by ENTER):</p>

<pre>
source("Coverage.R")
</pre>

<script src="https://gist.github.com/CnrLwlss/5895076.js"></script>

<p>The script above produces the following (thinned) plot of coverage along chromosome 11 in this experiment:</p>

<a href="Chr11.pdf"><img src="Chr11.png" title="Coverage along human chromosome 11 for a sample genome from the 1,000 genomes project at the EBI." width = 1000px></a>

<p>If you have your own .bam file you can use the script above to visualise coverage by copying it to your working directory, changing the sname and chrname variables in the "showReads.R" text file you have just created and running the script again.</p>

<h3>Human Mitochondrial DNA</h3>

<p>By writing a more complicated <a href="https://gist.github.com/CnrLwlss/5910629">example script</a>, also using ShortRead, we can visualise coverage along the circular, <a href="https://en.wikipedia.org/wiki/Mitochondrial_DNA">mitochondrial genome</a> during a <a href="http://en.wikipedia.org/wiki/ChIP-sequencing">ChIP-seq</a> experiment.  The script reads in a text file for annotating the <a href="https://en.wikipedia.org/wiki/Mitochondrial_DNA">mitochondrial genome</a> (which you can download <a href="http://lwlss.net/dwnlds/mtDNA_Colours.txt">here</a>) and a .bam alignment file (which you can download <a href="http://lwlss.net/dwnlds/IonXpress_018_R_2012_12_11_09_08_56_user_SN1-54_Auto_user_SN1-54_53.bam">here</a>) and uses these to produce
 plots like the one below.</p>
 
<a href="CHiP.svg"><img src="CHiP.png" title="Visualising coverage along the mitochondrial genome during a CHiP-seq experiment. Click for a zoomable, vector-graphics version of this plot." width = 600px></a>

 <p>As well as a working version of R and the ShortReads package, this mtDNA script requires two additional packages: plotrix and ecolitk, which can be installed by executing the following commands (and following the on-screen instructions) inside an R console:</p>
 
 <pre>
install.packages("plotrix")
source("http://bioconductor.org/biocLite.R")
biocLite("ecolitk")
 </pre>
 
 <p>To run the script, download the <a href="http://lwlss.net/dwnlds/mtDNA_Colours.txt">mtDNA_Colours.txt</a> file and the <a href="http://lwlss.net/dwnlds/IonXpress_018_R_2012_12_11_09_08_56_user_SN1-54_Auto_user_SN1-54_53.bam">.bam alignment file</a> to your working directory.  Copy the contents of the <a href="https://gist.github.com/CnrLwlss/5910629">example script</a> to a text file in your working directory, called "mtDNA.R" for example.  You can edit the <a href="http://lwlss.net/dwnlds/mtDNA_Colours.txt">mtDNA_Colours.txt</a> file to change gene labels for instance (though this file must remain in tab-delimited text format).  By editing the last line of the script, you can also change the name of the alignment file to be analysed, maximum number of reads for the plotted histogram (readmax), histogram colour (histColour), whether the number of reads is plotted on a log scale (logged=TRUE) or on a linear scale (logged=FALSE), as well as the text appearing in the centre of the circle (label).</p>
 
 <p>Finally, you can execute the script as above, by executing the following command in the windows GUI R console:</p>
 
 <pre>
source("mtDNA.R")
</pre>
	</article>
</body>	
</html>